src/coreclr/jit/lower.cpp

```diff
@@ -8871,15 +8688,101 @@ void Lowering::ContainCheckRet(GenTreeUnOp* ret)
 #endif // FEATURE_MULTIREG_RET
 }
 
+//------------------------------------------------------------------------
+// TryRemoveBitCast:
+//   Try to remove a bitcast node by changing its operand.
+//
+// Arguments:
+//    node - Bitcast node
+//
+// Returns:
+//   True if the bitcast was removed.
+//
+bool Lowering::TryRemoveBitCast(GenTreeUnOp* node)
```

Comment:
Curious if we run this folding at the import stage as well

Reply:
I don't think so, we only have some folding for bitcasts in morph, and there we only have the `IND/LCL_FLD` part of this.
Only reason I ended up adding this is because I hit some regressions in this PR due to the new bitcasts.